### PR TITLE
Completed lesson 1-4

### PR DESCRIPTION
**Pull request by camerongineer**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
-import {LIST_TITLES} from './data/data';
-import {buildList} from './utils/listUtils';
+import React, { useState } from "react";
+import TodoList from "./TodoList";
+import AddTodoForm from "./AddTodoForm";
 
 const App = () => {
-    const [newTodo, setNewTodo] = useState('');
+    const [todoList, setTodoList] = useState([]);
+    const addTodo = (newTodo) => setTodoList(todoList => [...todoList, newTodo]);
```
Comment (mattcarlotta) [inline src/App.js]: Nitpick (doesn't require changes):
Excellent usage of the state setter callback function! However, while this works as intended, there's some variable confusion between the `todoList` coming from `useState` and the `todoList` variable coming from the `setTodoList` callback function parameter. I'd recommend changing the variable naming within the callback function to create a visual separation of which variable is being used here:
```js
const addTodo = (newTodo) => setTodoList(prevTodoList => ([...prevTodoList, newTodo]));
```

File: src/AddTodoForm.js [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
-
+    
     return (
-        <form id={FORM_ID} onSubmit={handleAddTodo}>
+        <form onSubmit={handleAddTodo}>
             <label htmlFor={TITLE_ID}>Title:&nbsp;
-                <input id={TITLE_ID} name='title'/>
-                <button type='submit'>Add</button>
+                <input id={TITLE_ID} name="title" value={todoTitle} onChange={handleTitleChange}/>
+                <button type="submit">Add</button>
```
Comment (mattcarlotta) [inline src/AddTodoForm.js]: Requires changes:
This form submit button should not be a part of the `label`. The label can either wrap the `input` or be adjacent to it with an `id` attribute that references which input it's describing. However, the submit button is not a part of the label/input relationship and instead should reside outside of the label:
```jsx
return (
    <form onSubmit={handleAddTodo}>
        <label htmlFor={TITLE_ID}>Title:&nbsp;
            <input id={TITLE_ID} name="title" value={todoTitle} onChange={handleTitleChange}/>
         </label>
         <button type="submit">Add</button>
    </form>
);
```

**Review by mattcarlotta — CHANGES_REQUESTED** [review]
Overall the assignment looks excellent and you nearly aced it! I have one minor request change mentioned below.

File: src/AddTodoForm.js (hunk 2 above)
Comment (camerongineer) [reply src/AddTodoForm.js]: Thanks! I didn't realize that. I just made the changes.

**Review by mattcarlotta — APPROVED** [review]
Everything looks good to me, great job! 👍

**Pull request merged**